src/java/fr/paris/lutece/plugins/oauth2/jwt/JWTParser.java

```diff
@@ -44,22 +44,36 @@
  */
 public interface JWTParser
 {
+ 
     /**
-     * Extract Json Web Token from the token
-     *
-     * @param token
-     *            The token
+     *  Extract Json Web Token from the token
+     * @param <T>
+     * @param strCompactJwt
+     *            The compact jwt
      * @param clientConfig
      *            The client configuration
      * @param serverConfig
      *            The server configuration
-     * @param strStoredNonce
-     *            The stored nonce
+     * @param clazz
      * @param logger
      *            The logger
+     * @return <T>
      * @throws TokenValidationException
      *             if an error occurs
      */
-    void parseJWT( Token token, AuthClientConf clientConfig, AuthServerConf serverConfig, String strStoredNonce, Logger logger )
+    <T> T parseJWT( String strCompactJwt, AuthClientConf clientConfig, AuthServerConf serverConfig, Class<T> clazz, Logger logger  )
```

Comment:
Can you explain the rational for this change ? besides, this changes the behavior of this method, which previously set the IDToken into the Token (abeilt without properly documenting it). At the very least, documetation should be added to signify that where 1 call was previously necessary, two are now necessary